assets/javascripts/modules/SSO_links.js

```diff
@@ -19,25 +20,30 @@ require('jquery');
  * @returns {boolean}
  */
 module.exports = function(element, ssoUrl, ssoMethod) {
-  var $element,
-      payload,
-      clientSso,
-      serverSso,
-      destination,
-      newWindow,
-      winId;
+  var $element;
+  var payload;
+  var clientSso;
+  var serverSso;
+  var destination;
+  var winId;
+  var openInNewWindow;
+  var allowLinkCLickEvent = true;
   var elementHref;
+  var elementTarget;
   var useGet = ssoMethod === 'GET';
 
   if (element) {
     $element = $(element);
-    newWindow = !!$element.attr('target');
-    winId = element.id;
-    elementHref = element.href;
     clientSso = $element.data('sso') === true || $element.data('sso') === 'client';
     serverSso = $element.data('sso') === 'server';
 
     if (clientSso || serverSso) {
+
+      elementHref = element.href;
+      elementTarget = element.target;
+      openInNewWindow = elementTarget && elementTarget === '_blank';
```

Comment:
Ok - after a chat with @mattpepper this is the plan.

- quick fix for this now
```
openInNewWindow = elementTarget && ['_self', '_top', '_parent'].indexOf(elementTarget) === -1;
```

Then I will get a PR in for adding in to for adding support for a custom 'framename' [uk.gov.hmrc.urls.Link](https://github.com/hmrc/url-builder/blob/master/src/main/scala/uk/gov/hmrc/urls/Link.scala)
At present the only options in there are:  `true`, `false`, `self`, and `blank`

Then @mattpepper if you can bring Link.scala work in to the places you mentioned
[here](https://github.tools.tax.service.gov.uk/HMRC/agent-frontend/blob/master/app/views/includes/sidebarLinks.scala.html) and [here](https://github.tools.tax.service.gov.uk/HMRC/agent-frontend/blob/master/app/views/agentDashboard.scala.html)